lib/Parser.js

```diff
@@ -536,8 +536,10 @@ Parser.prototype.walkForOfStatement = function walkForOfStatement(statement) {
 
 // Declarations
 Parser.prototype.walkFunctionDeclaration = function walkFunctionDeclaration(statement) {
-	this.scope.renames["$" + statement.id.name] = undefined;
-	this.scope.definitions.push(statement.id.name);
+	if(statement.id) {
```

Comment:
Temporary fix

Ref https://github.com/estree/estree/issues/98

the error before was trying to access `name` of null (since id is null) for `export default function() {}`

Acorn: FunctionExpression
Babel: FunctionDeclaration + null id

---

Need to figure out why https://github.com/hzoo/webpack/blob/test-babylon/test/cases/parsing/issue-2570/fn.js

turns into 

### webpack + babylon

```js
/***/ (function(module, __webpack_exports__, __webpack_require__) {

"use strict";
function() { return [].slice.call(arguments); } // invalid

/***/ }),
```

### webpack master + acorn

```js
/***/ (function(module, __webpack_exports__, __webpack_require__) {

"use strict";
/* harmony default export */ __webpack_exports__["a"] = (function() { return [].slice.call(arguments); });

/***/ }),
```